rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.util;
+
+import co.rsk.RskContext;
+import co.rsk.config.NodeCliFlags;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+/**
+ * Created by Nazaret García on 21/01/2021
+ */
+
+public class PreflightChecksUtils {
+    private static final Logger logger = LoggerFactory.getLogger(PreflightChecksUtils.class);
+
+    private static final String[] SUPPORTED_JAVA_VERSIONS = {"1.8", "11"};
+
+    private static void checkSupportedJavaVersion() {
+        String javaVersion = SystemUtils.getPropertyValue("java.version");
+
+        if (javaVersion == null) {
+            throw new RuntimeException("Unable to detect Java version");
+        }
+
+        if (Arrays.stream(SUPPORTED_JAVA_VERSIONS).noneMatch(v -> isCompatibleVersion(javaVersion, v))) {
+            logger.error("Invalid Java version detected");
+            throw new RuntimeException(String.format("Invalid Java Version '%s'. Supported versions: %s", javaVersion, String.join(", ", SUPPORTED_JAVA_VERSIONS)));
```

Comment:
Why diff msgs? We can reuse the same msg for log/exception.

Reply:
I've updated that so we send the same message. 💪🏼